components/feature/prompts/src/main/java/mozilla/components/feature/prompts/dialog/SaveLoginDialogFragment.kt

```diff
@@ -289,77 +333,38 @@ internal class SaveLoginDialogFragment : PromptDialogFragment() {
     /**
      * Check current state then update view state to match.
      */
-    @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
-    fun update() = view?.toScope()?.launch(IO) {
-        val login = Login(
-            guid = guid,
-            origin = origin,
-            formActionOrigin = formActionOrigin,
-            httpRealm = httpRealm,
-            username = username,
-            password = password
-        )
-
-        try {
-            validateStateUpdate?.cancelAndJoin()
-        } catch (cancellationException: CancellationException) {
-            Logger.error("Failed to cancel job", cancellationException)
-        }
-
-        var validateDeferred: Deferred<Result>?
-        validateStateUpdate = launch validate@{
-            val validationDelegate =
-                feature?.loginValidationDelegate ?: return@validate
-            // We only want to fetch this list once. Since we are ignoring username the results will
-            // not change when user changes username or password fields
-            if (potentialDupesList == null) {
-                this@SaveLoginDialogFragment.potentialDupesList = CopyOnWriteArrayList(
-                    validationDelegate.getPotentialDupesIgnoringUsernameAsync(
-                        login
-                    ).await()
-                )
-            }
-            // Passing a copy of this potential dupes list, not a reference, so we know for certain
-            // that what shouldUpdateOrCreateAsync is using can't change.
-            validateDeferred = validationDelegate.shouldUpdateOrCreateAsync(
-                login,
-                potentialDupesList?.toList()
+    private fun update() {
+        // Figure out if we're going to be adding a login or updating an existing one
+        val loginToUpdate = findLoginToUpdate()
```

Comment:
`findLoginToUpdate` can end up invoking `update` via `fetchLoginsForOriginIfNeeded`. You're one slip-up  away from a non-obvious infinite loop :)

Reply:
This was from the fear I mentioned earlier about not being sure how to reliably start up the `loginsForOrigin` deferred, so I added this as a just in case `onCreateView()` hadn't already started it up.  I'll make sure to clean this up for the next pass.